src/Microsoft.AspNet.Mvc.Razor/Compilation/CompilationResult.cs

```diff
@@ -13,119 +11,49 @@ namespace Microsoft.AspNet.Mvc.Razor
     /// </summary>
     public class CompilationResult
     {
-        private Type _type;
-
         /// <summary>
-        /// Creates a new instance of <see cref="CompilationResult"/>.
+        /// Creates a new instance of <see cref="CompilationResult"/> that represents a success in compilation.
         /// </summary>
-        protected CompilationResult()
+        /// <param name="type">The compiled type.</param>
+        public CompilationResult([NotNull] Type compiledType)
```

Comment:
FYI generally agree w/ move from static creation methods to direct use of the constructors.  but this change seems like cleanup unrelated to this PR and it hides the important bits.